### PR TITLE
Add fields for variants

### PR DESCRIPTION
**Pull request by rgrinberg**
This is an attempt to add the variants fields (while still keeping them as no-ops) under the variants extension. Honestly, I'm not entirely sure how to make this work because extensions really just parse their own stanzas. While this is an extension to an existing language (which fits our version paradigm more).

@diml any ideas on how to make it work?

### CODE REVIEW

File: test/blackbox-tests/test-cases/variants/variants-using/dune-project [hunk 1]
```diff
@@ -0,0 +1,3 @@
+(lang dune 1.1)
+
+(using variants 1.0)
```
Comment (emillon) [inline test/blackbox-tests/test-cases/variants/variants-using/dune-project]: Maybe a simple way to declare intent that this is not completely stable is to use a `0.x` version here.

**Comment by diml** [comment]
Note that we already have such as case for menhir: the `menhir` stanza is controlled by the `menhir` extension, however the `flags` field of the menhir stanza is controlled by the `dune` language. The way it is done in this PR seems fine. The one thing that's missing is automatic edition of the `dune-project` file: when the user starts using a field that belong to the `variants` extension, dune should automatically adds `(using variants x.y)` to the `dune-project` file, just like it does it for the `menhir` stanza.

File: src/variant.mli [hunk 2]
```diff
@@ -1,17 +0,0 @@
-(** Library variants *)
```
Comment (diml) [inline src/variant.mli]: Why is this file gone?

**Comment by diml** [comment]
Can you also rename the extension so that it's clear user shouldn't play with it for now? i.e.:

```
(using IN_DEVELOPMENT_DO_NOT_USE!!_variants 0.0)
```

File: src/variant.mli (hunk 2 above)
Comment (rgrinberg) [reply src/variant.mli]: Accident. I brought it back.

**Comment by rgrinberg** [comment]
@diml I renamed it something more threatening.

File: test/blackbox-tests/test-cases/variants/run.t [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Variant feature is auto enabled when virtual_modules is used
+
+  $ dune build --root variants-without-using
+  File "dune", line 3, characters 1-25:
+   (virtual_modules foobar))
+   ^^^^^^^^^^^^^^^^^^^^^^^^
+  Error: 'virtual_modules' is only available since version 0.1 of experimental variants feature
```
Comment (diml) [inline test/blackbox-tests/test-cases/variants/run.t]: This sentence is odd, it should be "of the experimental variants features"

**Comment by diml** [comment]
Ok, looks good except for the missing `the`

**Comment by rgrinberg** [comment]
Done

**Pull request merged**